### PR TITLE
Update project structure, update some interfaces, separateQueryAndCursor tests

### PR DESCRIPTION
**Pull request by NikitaShkaruba**

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -136,8 +135,7 @@ const quotesRegex = /^'(.*)'$/;
 export function parseQuery<
     L extends LexerType,
     P extends ParserType,
-    V extends AbstractParseTreeVisitor<{}>,
-    S extends ISymbolTableVisitor & V,
```
Comment (NikitaShkaruba) [inline src/index.ts:150]: We don't need this extra V type

File: CODE_CONVENTIONS.md [hunk 2]
```diff
@@ -7,7 +7,7 @@ You can always link those rules in your PR without hesitation if you see that th
 
 - Tests should be put in `src/tests/shared` folder if logic is shared at least for 2 dialects, all unique logic should be tested in corresponding `src/tests/{dialect}` folder
 - In order to test logic for multiple dialects at the same time, you should use lib functions `groupParseSqlWithCursor`, `groupParseSqlWithoutCursor`, `groupParseSql`
-- All tests with cursor (expect when testing with newlines) should be written using `groupParseSqlWithCursor` or `parse{dialect}QueryWithCursor` functions. This will help us avoid passing cursor position to functions, and instead will calculate cursor position based on the pipe symbol `|`. E.g. `groupParseSqlWithCursor('SELECT |')` will calculate cursor position to be `{line: 1, column: 8}` and pass the correct query to the actual autocomplete function. These types of queries shouldn't include newlines.
+- All tests with cursor (expect when testing with newlines) should be written using `groupParseSqlWithCursor` or `parse{dialect}QueryWithCursor` functions. This will help us avoid passing cursor position to functions and make tests more readable
```
Comment (NikitaShkaruba) [inline CODE_CONVENTIONS.md:10]: I think that this information is better suited as a comment to the function

File: src/tests/shared/lib.test.ts [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    expect(() => {
+        separateQueryAndCursor('SELECT *\n FROM |')
+    }).toThrow(/Newline characters not allowed/);
+});
+
+test('should not work without cursor', () => {
+    expect(() => {
+        separateQueryAndCursor('SELECT * FROM ')
+    }).toThrow(/Cursor not provided/);
+});
```
Comment (NikitaShkaruba) [inline src/tests/shared/lib.test.ts:67]: @roberthovsepyan, I think when you were writing this code, you were testing it somehow anyway. But then it just got dropped. Instead we should put it to tests

File: src/tests/shared/lib.ts [hunk 4]
```diff
@@ -74,12 +74,17 @@ export function groupParseSqlWithoutCursor(
...
+// It adapts human-readable input to c3 readable input, making tests very readable.
+// Otherwise, we'd need to manually count line and column positions ourselves, which is very inconvenient for writing tests.
 export function separateQueryAndCursor(query: string): [string, CursorPosition] {
     if (lineSeparatorRegex.test(query)) {
         throw new Error(`Newline characters not allowed, but present in query ${query}`);
     }
 
-    const [queryBeforeCursor, queryAfterCursor, ...excessQueries] = query.split('|');
+    const cursorSymbol = '|';
+    const [queryBeforeCursor, queryAfterCursor, ...excessQueries] = query.split(cursorSymbol);
```
Comment (NikitaShkaruba) [inline src/tests/shared/lib.ts:87]: This makes it a little bit more readable

File: src/autocomplete/clickhouse/clickHouseAutocomplete.ts [hunk 5]
```diff
@@ -6,7 +6,7 @@ import {
     AutocompleteParseResult,
     ISymbolTableVisitor,
     AutocompleteData,
-    TableSuggestion
+    TableOrViewSuggestion
```
Comment (NikitaShkaruba) [inline src/autocomplete/clickhouse/clickHouseAutocomplete.ts:9]: In our main repository we separate those two entities, let's do it here too

**Review by NikitaShkaruba — COMMENTED** [review]
Some useful comments for the reviewers

**Review by roberthovsepyan — APPROVED** [review]

**Pull request merged**